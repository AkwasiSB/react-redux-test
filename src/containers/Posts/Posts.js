import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { handleFetchPosts } from '../../redux/actions/postActions';

export const PostItem = React.memo(function PostItem({ post }) {
	return (
		<div>
			<h3>{post.title}</h3>

			<p>{post.body}</p>
		</div>
	);
});

PostItem.propTypes = {
	post: PropTypes.shape({
		userId: PropTypes.number,
		id: PropTypes.number,
		title: PropTypes.string.isRequired,
		body: PropTypes.string.isRequired,
	}).isRequired
};

class Posts extends React.PureComponent {
	static propTypes = {
		fetchPosts: PropTypes.func.isRequired,
		posts: PropTypes.array.isRequired,
	}

	componentDidMount() {
		this.props.fetchPosts();
	}

	render() {
		return (
			<div>
				{this.props.posts.map(post => (
					<PostItem key={post.id} post={post} />
				))}
			</div>
		);
	}
}

function mapStateToProps({ postsData }) {
	return {
		posts: postsData.posts
	};
}

function mapDispatchToProps(dispatch) {
	return {
		fetchPosts: () => dispatch(handleFetchPosts())
	};
}

export default connect(mapStateToProps, mapDispatchToProps)(Posts);
