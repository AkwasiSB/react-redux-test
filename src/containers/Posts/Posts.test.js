import Posts, { PostItem } from './Posts';

describe('Rendering Posts made of PostItems', () => {
    it('it should render Posts', () => {
        const postsWrapper = shallow(<Posts posts={[]} fetchPosts={() => {}} />);

        expect(postsWrapper).toMatchSnapshot();
    });

    it('it should render PostItem', () => {
        const postData = {
            id: 1,
            user_id: '382980',
            title: 'Hey!',
            body: 'Hey buddy I am rendering cool stuff'
        };

        const postItemWrapper = shallow(<PostItem post={postData} />);

        expect(postItemWrapper).toMatchSnapshot();
    });
});
