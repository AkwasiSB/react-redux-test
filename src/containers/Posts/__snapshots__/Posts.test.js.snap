// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Posts made of PostItems it should render PostItem 1`] = `
<div>
  <h3>
    Hey!
  </h3>
  <p>
    Hey buddy I am rendering cool stuff
  </p>
</div>
`;

exports[`Rendering Posts made of PostItems it should render Posts 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
